TMS.Net07.Homework.Calculation/TMS.Net07.Homework.Calculation___hard/Program.cs

```diff
@@ -0,0 +1,126 @@
+                        Console.WriteLine($"{enterValue[0]} / {enterValue[1]} = {result}");
+                    }
+                    catch 
+                    {
+                        Console.WriteLine($"{errorMessage}");
+                    }
+                }
+                //multiplication
+                if (value == $"{enterValue[0]}*{enterValue[1]}")
+                {
+                    result = enterValue[0] * enterValue[1];
+                    Console.WriteLine($"{enterValue[0]} * {enterValue[1]} = {result}");
+                }
+                //remainder
+                if (value == $"{enterValue[0]}%{enterValue[1]}")
+                {
+                    result = enterValue[0] % enterValue[1];
+                    Console.WriteLine($"{enterValue[0]} % {enterValue[1]} = {result}");
+                }
+                //square
+                if (value == $"sqr{enterValue[1]}")
+                {
+                    result = enterValue[1] * enterValue[1];
+                    Console.WriteLine($"sqr{enterValue[1]} = {result}");
+                }
+                //square root
+                if (value == $"sqrt{enterValue[1]}")
+                {
+                    result = 1;
+                    for (double i = 1, check = 1; enterValue[1] - 0.01 > check && check < enterValue[1] + 0.01; i += 0.01)
+                    {
+                        check = i * i;
+                        result = i;
+                    }
+                    Console.WriteLine($"sqrt{enterValue[1]} = {result}");
+                }
+                //exponentiation
+                if (value == $"{enterValue[0]}pow{enterValue[1]}")
+                {
+                    double i = 1;
+                    double nextValue = enterValue[0];
+                    do
+                    {
+                        result = nextValue * enterValue[0];
+                        nextValue = result;
+                        i++;
+                    }
+                    while (i != enterValue[1]);
+                    Console.WriteLine($"{enterValue[0]} pow {enterValue[1]} = {result}");
+                }
+            }
+        }
+        //method for parsing enter string value in double for calculation;
+        static double[] parseEnterValue (string value)
+        {
+            string[] operationSeparator = { "+", "-", "/", "*", "%", "sqrt", "sqr", "pow" };
+            try
+            {
+                string[] enterValueString = value.Split(operationSeparator, StringSplitOptions.None);
+                double[] enterValueDouble = new double[enterValueString.Length];
+                for (int i = 0; i < enterValueString.Length; i++)
+                {
+                    {
+                        double.TryParse(enterValueString[i], out enterValueDouble[i]);
+                    }
+                }
+                return enterValueDouble;
+            }
+            catch
+            {
+                return null;
```

Comment:
In the `try` section there are no commands that would throw an exception which we could catch here. To make this logic work, we should replace `double.TryParse` with `double.Parse` which throws an exception if the input is wrong. `TryParse` just return `false` if the input is wrong.

Reply:
Also, for the future, I would suggest returning an empty array instead of null to avoid NullReferenceException